src/fparser/common/readfortran.py

```diff
@@ -955,6 +973,18 @@ def comment_item(self, comment, startlineno, endlineno):
         """
         return Comment(comment, (startlineno, endlineno), self)
 
+    def cpp_directive_item(self, line, startlineno, endlineno):
+        """ Construct :py:class:`CppDirective` item.
+
+        :param str line: String containing the text of a single or \
+        multi-line preprocessor directive
+        :param int startlineno: Start line number of the directive from \
+        the input source.
+        :param int endlineno: End line number of the directive from \
+        the input source.
+        """
```

Comment:
I leave a blank line before the last 3 quotes line, as you have done for CppDirective. It's not something that is enforced but it would be good to be consistent.